src/apps/api/tests/test_submissions.py

```diff
@@ -35,4 +52,75 @@ def test_making_a_submission_checks_if_you_are_a_participant(self):
         assert resp.status_code == 403
         assert "You do not have access to this competition to make a submission" in resp.data["detail"]
 
-    # TODO: Invalid secret screams?
+    def test_trying_to_change_submission_without_secret_raises_permission_denied(self):
+        url = reverse('submission-detail', args=(self.existing_submission.pk,))
+        # As anonymous user
+        resp = self.client.patch(url, {"status": Submission.FINISHED})
+        assert resp.status_code == 400
+        assert "Secret: (None) not a valid UUID" in str(resp.content)
+        assert Submission.objects.filter(pk=self.existing_submission.pk, status=Submission.SUBMITTED)
+
+        # As superuser (bad secret)
+        self.client.force_login(self.creator)
+        resp = self.client.patch(url, {"status": Submission.FINISHED, "secret": '7df3600c-aa6d-41c5-bbc8-bbe91f42d875'})
+        assert resp.status_code == 403
+        assert resp.data["detail"] == "Submission secrets do not match"
+        assert Submission.objects.filter(pk=self.existing_submission.pk, status=Submission.SUBMITTED)
+
+        # As anonymous user with secret
+        self.client.logout()
+        resp = self.client.patch(url, {"status": Submission.FINISHED, "secret": self.existing_submission.secret})
+        assert resp.status_code == 200
+        assert Submission.objects.filter(pk=self.existing_submission.pk, status=Submission.FINISHED)
+
+    def test_cannot_delete_submission_you_didnt_create(self):
+        url = reverse('submission-detail', args=(self.existing_submission.pk,))
+
+        # As anonymous user
+        resp = self.client.delete(url)
+        assert resp.status_code == 403
+        assert resp.data["detail"] == "Cannot interact with submission you did not make"
+
+        # As regular user
+        self.client.force_login(self.other_user)
+        resp = self.client.delete(url)
+        assert resp.status_code == 403
+        assert resp.data["detail"] == "Cannot interact with submission you did not make"
+
+        # As user who made submission
```

Comment:
should we also test superuser can delete submissions?

Reply:
Good call, they weren't able to. added that, fixed test